src/analysis/rust_type.rs

```diff
@@ -191,11 +196,59 @@ fn rust_type_full(
             }
         }
         Custom(library::Custom { ref name, .. }) => Ok(name.clone()),
+        Function(ref f) => {
+            let full_name = type_id.full_name(&env.library);
+            if full_name == "Gio.AsyncReadyCallback" {
```

Comment:
Elsewhere you looked at the c_type, here you look at something else. Seems inconsistent

Reply:
We don't have the parameter here so different check. :)